### PR TITLE
#33902 Configureren taak voor maken Zaakbesluit

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: besluit/src/main/kotlin/com/ritense/besluit/security/config/BesluitConfigHttpSecurityConfigurer.kt [hunk 1]
```diff
@@ -16,16 +16,18 @@
...
 import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import javax.ws.rs.HttpMethod.GET
 
 class BesluitConfigHttpSecurityConfigurer : HttpSecurityConfigurer {
 
     override fun configure(http: HttpSecurity) {
         try {
             http.authorizeRequests()
-//                .antMatchers(GET, "/api/object/sync/config/{id}").hasAuthority(ADMIN)
+                .antMatchers(GET, "/api/besluit/types").hasAuthority(ADMIN)
```
Comment (ivo-ritense) [inline besluit/src/main/kotlin/com/ritense/besluit/security/config/BesluitConfigHttpSecurityConfigurer.kt:30]: besluittypes is 1 word.

File: besluit/src/main/kotlin/com/ritense/besluit/web/rest/impl/BesluitResource.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.ritense.besluit.web.rest.response.BesluitTypeResponse
+import com.ritense.openzaak.service.impl.ZaakTypeService
+import org.springframework.http.ResponseEntity
+
+class BesluitResource(
+    val zaakTypeService: ZaakTypeService,
+) : BesluitResource {
+
+    override fun getBesluitTypen(): ResponseEntity<List<BesluitTypeResponse>> {
+        val besluitTypes = zaakTypeService.getBesluitTypes().results
```
Comment (ivo-ritense) [inline besluit/src/main/kotlin/com/ritense/besluit/web/rest/impl/BesluitResource.kt:29]: This mapping should happen in the service itself

File: openzaak/src/main/kotlin/com/ritense/openzaak/service/ZaakTypeService.kt [hunk 3]
```diff
@@ -17,10 +17,13 @@
...
 
 import com.ritense.openzaak.service.impl.model.ResultWrapper
+import com.ritense.openzaak.service.impl.model.catalogi.BesluitType
 import com.ritense.openzaak.service.impl.model.catalogi.ZaakType
 
 interface ZaakTypeService {
 
     fun getZaakTypes(): ResultWrapper<ZaakType>
 
+    fun getBesluitTypes(): ResultWrapper<BesluitType>
```
Comment (ivo-ritense) [inline openzaak/src/main/kotlin/com/ritense/openzaak/service/ZaakTypeService.kt:27]: Though i don't think it's bad to combine this, the class name doesn't match the method added.

File: openzaak/src/main/kotlin/com/ritense/openzaak/domain/mapping/impl/Operation.kt [hunk 4]
```diff
@@ -19,5 +19,6 @@ package com.ritense.openzaak.domain.mapping.impl
 enum class Operation {
     CREATE_ZAAK,
     SET_RESULTAAT,
-    SET_STATUS
+    SET_STATUS,
+    CREATE_BESLUIT,
```
Comment (ivo-ritense) [inline openzaak/src/main/kotlin/com/ritense/openzaak/domain/mapping/impl/Operation.kt:23]: Is this tested somewhere?

Comment (Klaas-Ritense) [reply openzaak/src/main/kotlin/com/ritense/openzaak/domain/mapping/impl/Operation.kt]: i removed this

File: openzaak/src/main/kotlin/com/ritense/openzaak/service/ZaakTypeService.kt (hunk 3 above)
Comment (Klaas-Ritense) [reply openzaak/src/main/kotlin/com/ritense/openzaak/service/ZaakTypeService.kt]: I moved it to a new client

File: besluit/src/main/kotlin/com/ritense/besluit/web/rest/impl/BesluitResource.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply besluit/src/main/kotlin/com/ritense/besluit/web/rest/impl/BesluitResource.kt]: I moved the mapping to a new service

File: besluit/src/main/kotlin/com/ritense/besluit/security/config/BesluitConfigHttpSecurityConfigurer.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply besluit/src/main/kotlin/com/ritense/besluit/security/config/BesluitConfigHttpSecurityConfigurer.kt]: change it to /besluittype

**Pull request merged**